### DISCUSSION TITLE
Set default anonymous level for channel seeding/downloading/updating

### DISCUSSION THREAD

**Discussion by CrocodileCroco** [Tribler/tribler#6100] [Q&A]
When i download (subscribe) a channel, there is no option to do it anonymously, same for updating channel, you always need to go to the downloads tab and make it anonymous

You should be able to seed/dl/update channels anonymously

**Comment by ichorid** [comment]
Anonymous seeding of channels is pointless, because of how channel gossip works: you gossip subscribed channels, so **everyone** around you know what channels you are subscribed to. An open Channels system incentivizes people to only share links to legal and quality content.

Tribler is about providing *links* to information (i.e. infohashes), and *means* of downloading it without fear of spying by 3rd parties (e.g. nosy ISPs).

So, "anonymous channels" not gonna happen in the foreseeable future.

**Reply by lucy2812** [reply]
Does "an open Channels system" create a liability and risk to those that subscribe?

**Reply by ichorid** [reply]
If your jurisdiction prohibits sharing *links* to information (such as infohashes), that's definitely a liability. The channel subscribers could be trivially discovered by joining the channel torrent swarm and recording the participants' IP addresses.

Anonymizing channels would be a trivial task, but we're **not** going to do it until we install the means for the community to police itself and clean itself of illegal and/or useless content.

**Reply by philhoefer** [reply]
> If your jurisdiction prohibits sharing _links_ to information (such as infohashes), that's definitely a liability.

Afaik that is true in at least the EU (Article 3.2 of the [EU Copyright Directive](https://eur-lex.europa.eu/legal-content/EN/TXT/HTML/?uri=CELEX:32001L0029&from=EN)). There is a [judgement from the ECJ](https://eur-lex.europa.eu/legal-content/en/TXT/PDF/?uri=uriserv%3AOJ.C_.2017.277.01.0008.01.ENG) about this.
I don't like this one bit but as it stands this makes usage of Tribler a potential liability for a huge chunk of people. 😢

**Reply by ichorid** [reply]
Well, the Web itself falls under that judgement, along with Google as the indexing engine :shrug:

**Reply by ichorid** [reply]
Anyway, the project's primary goal is to develop methods of organizing people into [a market capable of self-governance](https://github.com/Tribler/tribler/issues/1). We're **not** building a darknet. Instead, we're building [a "lightnet"](https://github.com/Tribler/tribler/issues/4486), so to speak. It is up to governments to enforce people using real identities on the Web.

**Reply by philhoefer** [reply]
As much as I agree that the legislation and its interpretation by the ECJ are more or less illogical to any technically-minded person, it still is the current law. And no, what Google does is not the same.
Vexingly, even publishing a hyperlink to a page that illegally contains unlicensed content is considered illegal in EU law. According to [another judgement,](https://curia.europa.eu/juris/document/document.jsf?text=&docid=183124&pageIndex=0&doclang=EN&mode=lst&dir=&occ=first&part=1&cid=8887695) the point is if the person that provides the link `did not know or could not reasonably have known the illegal nature of the publication of those works on that other website`.
Google can reasonably state that they did not know about the illegality of a link they indexed while crawling the web. Google regularily deletes those links. A tribler user who manually subscribed to a channel that clearly has such content can't really argue that they did not know about it.

The point that I'm trying to accomplish here is that this could become a problem for a lot of Tribler users as soon as tribler gains a larger user base. As soon as it has that user base it would be trivial for immoral lawyers to get a list of subscribers to a channel and sue them en masse.
I'd wager that most Trible users are not aware this kind of legal attack is possible.

**Reply by philhoefer** [reply]
> 
> Anyway, the project's primary goal is to develop methods of organizing people into [a market capable of self-governance](https://github.com/Tribler/tribler/issues/1). We're **not** building a darknet. Instead, we're building [a "lightnet"](https://github.com/Tribler/tribler/issues/4486), so to speak. It is up to governments to enforce people using real identities on the Web.

If this is out of scope for Tribler (which it seems like from your reply) then I guess a warning message that subscribing to a channel also unanonymously shares that channel would be helpful to users.

**Reply by ichorid** [reply]
>did not know or could not reasonably have known the illegal nature of the publication of those works on that other website.

We're going to decouple _seeding_ the channel from _subscribing_ to and _voting_ for it by the end of this year. The reasons are purely technical (scalability, availability, etc.). Coincidentally, this will make it impossible to tell if the user is sharing the channel _willingly_. This will require a major refactoring of the Channels, though.

**Comment by Snake883** [comment]


**Reply by ichorid** [reply]
> If I understand correctly, if I create my own channel, then anyone can associate my IP address with my content?

Yes, because your personal channel is signed with the private key of your Tribler instance. Though, you can always create a channel and then swap your Tribler private key for another one. In that case, it will look like you subscribed and downloaded the channel (instead of created it).

>If I subscribe to a channel, anyone can relate my IP address with the content of that channel?

Yes.
